File: src/DMusicBot/Dtos/PlayerUpdatedDto.cs
using Lavalink4NET.Players;

namespace DMusicBot.Dtos;

public record struct PlayerUpdatedDto
{
    public bool UpdateQueue { get; init; }
    public TrackDto? CurrentTrack { get; init; }
    public int PositionInSeconds { get; init; }
    public TrackDto[] Queue { get; init; }
    public PlayerState State { get; init; }
}

File: src/frontend/src/dtos/PlayerUpdatedDto.ts
import {TrackDto} from "@/dtos/TrackDto";
import {PlayerState} from "@/datatypes/PlayerState";

export interface PlayerUpdatedDto {
    updateQueue: boolean;
    currentTrack: TrackDto;
    positionInSeconds: number;
    queue: TrackDto[];
    state: PlayerState;
}